mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotation.java

```diff
@@ -0,0 +1,128 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.model;
```

Comment:
Ok, so I think the overall design here is wrong.

You copied the overall ACL design, and manage the annotations as a wall.

This is needed for the ACLs. Because of simplification logic. (consider you give the right on a mailbox, and then rmove me the right in an other ACL, then removing the first ACL is needed). Several ACL entries might be correlated. Hence the need to manage it as a wall.

For the annotations, they are not correlated, so you can manage them in a simplest way : 

Data

 ```
MailboxAnnotation {
 private final String key;
 private final String value;
 // ...
}
```

Because no resolution is needed you can go for a simple CRUD (single annotation, multiple annotations)